File: example/src/main/java/com/iheartradio/example/typeadapters/UpperCaseStringTypeAdapter.java
package com.iheartradio.example.typeadapters;

import android.view.ViewGroup;

import com.iheartradio.example.data.UpperCaseStringData;
import com.iheartradio.example.viewholders.ListItemOneViewHolder;
import com.iheartradio.multitypeadapter.TypeAdapter;

import java.util.List;

/**
 * Created by Jonathan Muller on 3/7/17.
 */

public class UpperCaseStringTypeAdapter extends TypeAdapter<UpperCaseStringData, ListItemOneViewHolder> {

    @Override
    public boolean isMyData(Object data) {
        return data instanceof UpperCaseStringData;
    }

    @Override
    public ListItemOneViewHolder onCreateViewHolder(final ViewGroup parent) {
        return ListItemOneViewHolder.create(parent);
    }

    @Override
    public boolean isDataEqual(final UpperCaseStringData data1, final UpperCaseStringData data2) {
        if (data1 == null || data2 == null) {
            return false;
        }
        return data1.getData().equals(data2.getData());
    }

    @Override
    public void onBindViewHolder(ListItemOneViewHolder viewHolder, UpperCaseStringData data, final List<Object> payloads) {
        viewHolder.bind(data);
    }
}

File: example/src/main/java/com/iheartradio/example/typeadapters/UpperCaseStringTypeAdapter.kt
package com.iheartradio.example.typeadapters

import android.view.ViewGroup

import com.iheartradio.example.data.UpperCaseStringData
import com.iheartradio.example.viewholders.ListItemOneViewHolder
import com.iheartradio.multitypeadapter.TypeAdapter

/**
 * Created by Jonathan Muller on 3/7/17.
 */

class UpperCaseStringTypeAdapter : TypeAdapter<UpperCaseStringData, ListItemOneViewHolder>() {

    override fun isMyData(data: Any): Boolean {
        return data is UpperCaseStringData
    }

    override fun onCreateViewHolder(parent: ViewGroup): ListItemOneViewHolder {
        return ListItemOneViewHolder.create(parent)
    }

    override fun isDataEqual(data1: UpperCaseStringData, data2: UpperCaseStringData): Boolean {
        return data1.data == data2.data
    }

    fun onBindViewHolder(viewHolder: ListItemOneViewHolder, data: UpperCaseStringData, payloads: List<Any>) {
        viewHolder.bind(data)
    }
}
